core/controllers/classifier.py

```diff
@@ -123,3 +129,61 @@ def post(self):
         classifier_services.mark_training_job_complete(job_id)
 
         return self.render_json({})
+
+
+def update_failed_jobs(job_models):
+    for job_model in job_models:
+        classifier_services.mark_training_job_failed(job_model.id)
+
+
+def fetch_training_data(exp_id, state):
```

Comment:
But, do you know why we ever wanted to populate training data in the controller in the first place? I don't see why its required.